src/Symfony/Component/HttpFoundation/Tests/ParameterBagTest.php

```diff
@@ -378,6 +378,30 @@ public function testGetEnumThrowsExceptionWithInvalidValueType()
 
         $this->assertNull($bag->getEnum('invalid-value', FooEnum::class));
     }
+
+    public function testGetStringOrNull(): void
+    {
+        $bag = new ParameterBag(['valid-value' => 'foo']);
+
+        $this->assertSame('foo', $bag->getStringOrNull('valid-value'));
+        $this->assertNull($bag->getStringOrNull('invalid-key'));
+    }
+
+    public function testGetIntOrNull(): void
+    {
+        $bag = new ParameterBag(['valid-value' => 1]);
```

Comment:
An assertion with an invalid value is necessary, i.e. with a value that cannot be converted to int. Same for Boolean.

Reply:
I've changed it to using native `getInt` calls etc. so behaviour will be same in both cases.